pkg/webui/console/store/middleware/application.js

```diff
@@ -66,8 +66,57 @@ const getApplicationApiKeyDataLogic = createLogic({
   },
 })
 
+const getApplicationCollaboratorsLogic = createLogic({
+  type: application.GET_APP_COLLABORATORS_LIST,
+  async process ({ getState, action }, dispatch, done) {
+    const { id } = action
+    try {
+      const res = await api.application.collaborators.list(id)
+      const collaborators = res.collaborators.map(function (collaborator) {
+        const { ids, ...rest } = collaborator
+        const isUser = !!ids.user_ids
+        const collaboratorId = isUser
+          ? ids.user_ids.user_id
+          : ids.organization_ids.organization_id
+
+        return {
+          id: collaboratorId,
+          isUser,
+          ...rest,
+        }
+      })
+
+      dispatch(
+        application.getApplicationCollaboratorsListSuccess(
+          id,
+          collaborators,
+          res.totalCount
+        )
+      )
+    } catch (e) {
+      dispatch(application.getApplicationCollaboratorsListFailure(id, e))
+    }
+
+    done()
+  },
+})
+
+const getApplicationCollaboratorDataLogic = createLogic({
+  type: application.GET_APP_COLLABORATOR,
```

Comment:
> This action name is confusing. It took me a while to realize that this action is not for retrieving a single collaborator. GET_APP_COLLABORATOR_PAGE_DATA ?

Well, it is exactly as I implemented it for the API Keys management.

> Also, instead of having an extra logic to orchestrate dispatches, consider adding the actions to the type property of the corresponding logics

Imo, it makes it harder to trace the action execution sequence. In this particular case, `GET_APP_COLLABORATOR_PAGE_DATA` will be scattered across `middleware/applications` and `middleware/application`while the current approach on can clearly see what middleware is triggered and what actions are dispatched in an 'atomic' fashion.